flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/SourceOperator.java

```diff
@@ -589,10 +589,6 @@ public void updateCurrentEffectiveWatermark(long watermark) {
     @Override
     public void updateCurrentSplitWatermark(String splitId, long watermark) {
         splitCurrentWatermarks.put(splitId, watermark);
-        if (currentMaxDesiredWatermark < watermark && !currentlyPausedSplits.contains(splitId)) {
-            pauseOrResumeSplits(Collections.singletonList(splitId), Collections.emptyList());
-            currentlyPausedSplits.add(splitId);
-        }
```

Comment:
Why are we removing this? If suddenly split wakes up and starts sending huge amount of records, lack of this code can cause problems.

edit: oh, you have reverted this change later, haven't you?

Reply:
Yes, I didn't remove this but I fixed a bug which @mas-chen pointed out which would pause the split when there is just a single split.